### ISSUE TITLE
dartfmt and dartanalyzer are conflicting when using linter rule always_put_control_body_on_new_line

### ISSUE THREAD

**Issue by apaatsio** [dart-lang/dart_style#808]
`dartfmt` formats code in a way that doesn't agree with the linter rule `always_put_control_body_on_new_line`. This creates quite an annoying problem where I cannot use `dartfmt` if I want to use the rule `always_put_control_body_on_new_line`, and I cannot use the said linter rule if I want to use `dartfmt`.

Here's an example. Linter finds no issues before formatting but finds one issue afterward.

## Log
```
$ cat analysis_options.yaml
linter:
  rules:
    - always_put_control_body_on_new_line

$ cat test.dart
void main() {
  if(true)
    return;
}

$ dartanalyzer test.dart 
Analyzing test.dart...
No issues found!

$ dartfmt -w test.dart 
Formatted test.dart

$ dartanalyzer test.dart 
Analyzing test.dart...
  lint • Separate the control structure expression from its statement at test.dart:2:13 • always_put_control_body_on_new_line
1 lint found.

$ cat test.dart
void main() {
  if (true) return;
}
```

## Versions

- dart: Dart VM version: 2.3.0-dev.0.1.flutter-cf4444b803 (Thu Apr 18 21:21:18 2019 +0000) on "linux_x64"
- dartfmt: 1.2.7
- dartanalyzer: 2.2.1-edge.cf4444b803a376bb1ad6442e1f09434a4d58167b

**Comment by apaatsio** [comment]
Somewhat related discussion: flutter/flutter#31448

**Comment by munificent** [comment]
dartfmt is [opinionated by design](https://github.com/dart-lang/dart_style/wiki/FAQ#why-cant-i-configure-it). It does it's best to format your code to one canonical style so that all code formatted using dartfmt is consistent.

The linter is not opinionated. Most lint rules do fit within dartfmt and "Effective Dart"'s guidelines, but some do not. If you choose to enable a lint rule that conflicts with dartfmt, then using dartfmt will produce lint errors.

The easiest fix in this case is to always put braces on your ifs. The lint is happy with braced ifs and dartfmt will format them the way you expect. It is only on unbraced ifs where you run into conflict.
